### PR TITLE
New container image tagging strategy

### PR DESCRIPTION
**Pull request by pperzyna**

### CODE REVIEW

File: .github/workflows/facebook.yml [hunk 1]
```diff
@@ -3,31 +3,52 @@ name: Build Facebook plug
 on:
   push:
     tags:
-      - 'facebook-v*.*.**'
+      - 'facebook-v**'
```
Comment (whiteshadow-gr) [inline .github/workflows/facebook.yml:6]: How do we differentiate between a dev plug and a normal plug?

File: .github/workflows/facebook.yml [hunk 2]
```diff
@@ -3,31 +3,52 @@ name: Build Facebook plug
...
+      - name: Container Image Push - Tag
+        if: startsWith(github.ref, 'refs/tags')
         run: |
-          docker build -t hubofallthings/$REPOSITORY $DATAPLUG/target/docker/stage/
-          docker login -u "${{ secrets.DOCKER_USERNAME }}" -p "${{ secrets.DOCKER_PASSWORD }}"
-          docker push hubofallthings/$REPOSITORY
+          docker tag hubofallthings/$DATAPLUG:$GITHUB_SHA hubofallthings/$DATAPLUG:$(echo $GITHUB_REF | sed -e "s/refs\/tags\///g")
+          docker push hubofallthings/$DATAPLUG:$(echo $GITHUB_REF | sed -e "s/refs\/tags\///g")
+          docker tag hubofallthings/$DATAPLUG:$GITHUB_SHA hubofallthings/$DATAPLUG:latest
+          docker push hubofallthings/$DATAPLUG:latest
```
Comment (whiteshadow-gr) [inline .github/workflows/facebook.yml:54]: I thought `latest` was to be removed?

File: .github/workflows/facebook.yml (hunk 1 above)
Comment (pperzyna) [reply .github/workflows/facebook.yml:6]: as far as I understand the dev version is just for the DEV deployment, but the code is the same as for PRD. So, in that case, there is no need to tag it separately or even keep it on separate repositories. 

You can always build a Facebook version, but if the version is broken or doesn't work well, just don't deploy it to PRD. 
Moreover, to inform the community which version (git tag) is stable, we can simply create a release on a particular git tag

File: .github/workflows/facebook.yml (hunk 2 above)
Comment (pperzyna) [reply .github/workflows/facebook.yml:54]: I would like to keep `latest` tag as an additional tag and use it only as a cache for future builds.

File: .github/workflows/fitbit.yml [hunk 3]
```diff
@@ -3,31 +3,52 @@ name: Build Fitbit plug
...
+      - name: Container Image Push - Tag
+        if: startsWith(github.ref, 'refs/tags')
         run: |
-          docker build -t hubofallthings/$REPOSITORY $DATAPLUG/target/docker/stage/
-          docker login -u "${{ secrets.DOCKER_USERNAME }}" -p "${{ secrets.DOCKER_PASSWORD }}"
-          docker push hubofallthings/$REPOSITORY
+          docker tag hubofallthings/$DATAPLUG:$GITHUB_SHA hubofallthings/$DATAPLUG:$(echo $GITHUB_REF | sed -e "s/refs\/tags\///g")
+          docker push hubofallthings/$DATAPLUG:$(echo $GITHUB_REF | sed -e "s/refs\/tags\///g")
+          docker tag hubofallthings/$DATAPLUG:$GITHUB_SHA hubofallthings/$DATAPLUG:latest
+          docker push hubofallthings/$DATAPLUG:latest
```
Comment (whiteshadow-gr) [inline .github/workflows/fitbit.yml:54]: Please add an empty line

File: .github/workflows/spotify.yml [hunk 4]
```diff
@@ -3,31 +3,52 @@ name: Build Spotify plug
...
+      - name: Container Image Push - Tag
+        if: startsWith(github.ref, 'refs/tags')
         run: |
-          docker build -t hubofallthings/$REPOSITORY $DATAPLUG/target/docker/stage/
-          docker login -u "${{ secrets.DOCKER_USERNAME }}" -p "${{ secrets.DOCKER_PASSWORD }}"
-          docker push hubofallthings/$REPOSITORY
+          docker tag hubofallthings/$DATAPLUG:$GITHUB_SHA hubofallthings/$DATAPLUG:$(echo $GITHUB_REF | sed -e "s/refs\/tags\///g")
+          docker push hubofallthings/$DATAPLUG:$(echo $GITHUB_REF | sed -e "s/refs\/tags\///g")
+          docker tag hubofallthings/$DATAPLUG:$GITHUB_SHA hubofallthings/$DATAPLUG:latest
+          docker push hubofallthings/$DATAPLUG:latest
```
Comment (whiteshadow-gr) [inline .github/workflows/spotify.yml:54]: New line please

File: .github/workflows/google-calendar.yml [hunk 5]
```diff
@@ -3,31 +3,52 @@ name: Build Google Calendar plug
...
+      - name: Container Image Push - Tag
+        if: startsWith(github.ref, 'refs/tags')
         run: |
-          docker build -t hubofallthings/$REPOSITORY $DATAPLUG/target/docker/stage/
-          docker login -u "${{ secrets.DOCKER_USERNAME }}" -p "${{ secrets.DOCKER_PASSWORD }}"
-          docker push hubofallthings/$REPOSITORY
+          docker tag hubofallthings/$DATAPLUG:$GITHUB_SHA hubofallthings/$DATAPLUG:$(echo $GITHUB_REF | sed -e "s/refs\/tags\///g")
+          docker push hubofallthings/$DATAPLUG:$(echo $GITHUB_REF | sed -e "s/refs\/tags\///g")
+          docker tag hubofallthings/$DATAPLUG:$GITHUB_SHA hubofallthings/$DATAPLUG:latest
+          docker push hubofallthings/$DATAPLUG:latest
```
Comment (whiteshadow-gr) [inline .github/workflows/google-calendar.yml:54]: New line please

**Review by whiteshadow-gr — APPROVED** [review]
Do you think we should include the slack action here as well? Just checking what are your thoughts about it

**Review by dataswifty — APPROVED** [review]

**Comment by dataswifty** [comment]
Does it make sense to keep using your personal Github account for these?

**Pull request merged**